instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py

```diff
@@ -0,0 +1,653 @@
+    input("...")
+
+    # to configure custom metrics
+    configuration = {
+        "system.memory.usage": ["used", "free", "cached"],
+        "system.cpu.time": ["idle", "user", "system", "irq"],
+        "system.network.io": ["trasmit", "receive"],
+        "runtime.memory": ["rss", "vms"],
+        "runtime.cpu.time": ["user", "system"],
+    }
+    SystemMetricsInstrumentor(config=configuration).instrument()
+
+API
+---
+"""
+
+import gc
+import os
+from platform import python_implementation
+from typing import Collection, Dict, Iterable, List, Optional
+
+import psutil
+
+from opentelemetry._metrics import get_meter
+from opentelemetry._metrics.measurement import Measurement
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.instrumentation.system_metrics.package import _instruments
+from opentelemetry.instrumentation.system_metrics.version import __version__
+from opentelemetry.sdk.util import get_dict_as_key
+
+
+class SystemMetricsInstrumentor(BaseInstrumentor):
+    # pylint: disable=too-many-statements
+    def __init__(
+        self,
+        labels: Optional[Dict[str, str]] = None,
+        config: Optional[Dict[str, List[str]]] = None,
+    ):
+        super().__init__()
+        if config is None:
+            self._config = {
+                "system.cpu.time": ["idle", "user", "system", "irq"],
+                "system.cpu.utilization": ["idle", "user", "system", "irq"],
+                "system.memory.usage": ["used", "free", "cached"],
+                "system.memory.utilization": ["used", "free", "cached"],
+                "system.swap.usage": ["used", "free"],
+                "system.swap.utilization": ["used", "free"],
+                # system.swap.page.faults: [],
+                # system.swap.page.operations: [],
+                "system.disk.io": ["read", "write"],
+                "system.disk.operations": ["read", "write"],
+                "system.disk.time": ["read", "write"],
+                # "system.filesystem.usage": [],
+                # "system.filesystem.utilization": [],
+                "system.network.dropped.packets": ["transmit", "receive"],
+                "system.network.packets": ["transmit", "receive"],
+                "system.network.errors": ["transmit", "receive"],
+                "system.network.io": ["trasmit", "receive"],
+                "system.network.connections": ["family", "type"],
+                "runtime.memory": ["rss", "vms"],
+                "runtime.cpu.time": ["user", "system"],
+            }
+        else:
+            self._config = config
+        self._labels = {} if labels is None else labels
+        self._meter = None
+
+    def instrumentation_dependencies(self) -> Collection[str]:
+        return _instruments
+
+    def _instrument(self, **kwargs):
+        meter_provider = kwargs.get("meter_provider")
+        self._meter = get_meter(
+            "io.otel.python.instrumentation.system_metrics",
```

Comment:
is this consistent with other insturmentation packages?

Reply:
i've reverted back to using `__name__` as per other instrumentations in this repo... we should consider if this is correct moving forward though.

Reply:
Issue added: https://github.com/open-telemetry/opentelemetry-python-contrib/issues/1035